### ISSUE TITLE
Actually use GlobalLocationProvider

### ISSUE THREAD

**Issue by nighelles** [remulasce/metroapp#256]
This is actually done, mostly. I implemented the interface, and pass in my location class. However, new code also seems to be breaking my code for getting arrivals, so that's still in progress.

**Comment by remulasce** [comment]
We should probably lock java_core until releaae

**Comment by nighelles** [comment]
Looks like issue is with tempSRDA not returning anything, breaking a log statement that looks at the first element.

**Comment by nighelles** [comment]
Did the internal format change? multiArrivalTrips is saying it has like 14 elements, but none of them have arrivals.

**Comment by remulasce** [comment]
Format of the what?

**Comment by nighelles** [comment]
The way trips are stored

**Comment by remulasce** [comment]
No?

**Comment by nighelles** [comment]
Seems odd that they'd just stop returning arrivals. That code didn't change at all.

**Comment by nighelles** [comment]
looks like SRDA is lying about how many multiarrivaltrips it has?

**Comment by nighelles** [comment]
WTF? I'm getting a tempSRDA that's just a string that says @"...".

What's the deal with that

**Comment by nighelles** [comment]
Ok, so this is odd. Looks like I'm getting a multiarrivaltrip that's actually just a trip. So it doesn't have a proper parentArrival, just a text parameter defaulting to "..."

Since I'm actually using Objective-C and not Java, it thinks it's a proper parentArrival, and dutifully returns the object at the correct memory location, which is the String that says "..."

**Comment by nighelles** [comment]
But I feel like trip should still declare parentArrival, because MultiArrivalTrip assumes it exists. So what's the deal with that, huh?

**Comment by nighelles** [comment]
No, ok it's declared in MultiArrivalTrip. But somehow I'm getting a normal Trip, rather than a multiArrivalTrip.

**Comment by nighelles** [comment]
maybe the statustrip thing is trying to get displayed? I thought I fixed that.

**Comment by nighelles** [comment]
Ah, I did fix it, on the iosdev branch. Which for some reason didn't get merged properly.

**Comment by nighelles** [comment]
Ugh, merged in that code, but now shit is whack because I don't understand why it's giving me empty multiarrivaltrips.

**Comment by nighelles** [comment]
I'm getting just "54855" as a description back from a multiarrival trip. That's odd.

**Comment by remulasce** [comment]
Fix your logging/println implementation.
